config/configload/error_handler.go

```diff
@@ -87,16 +87,15 @@ func newErrorHandlerContent(content *hcl.BodyContent) (map[string]struct{}, []ki
 			}
 
 			if k != errors.Wildcard && !errors.IsKnown(k) {
-				subjRange := block.DefRange
-				if len(block.LabelRanges) > 0 {
-					subjRange = block.LabelRanges[0]
+				if len(block.LabelRanges) == 0 {
+					block.LabelRanges = append(block.LabelRanges, block.DefRange)
```

Comment:
Does this modify a parsed config body in memory? Would the next caller see a modified range?
I prefer changing a local variable.

Reply:
So better have
```go
				subjRange := block.DefRange
				if len(block.LabelRanges) > 0 {
					subjRange = block.LabelRanges[0]
				}
				diag := &hcl.Diagnostic{
					Severity: hcl.DiagError,
					Summary:  fmt.Sprintf(...),
					Subject:  &subjRange,
				}
				return nil, nil, hcl.Diagnostics{diag}
```
in both cases?